x-pack/plugins/fleet/server/services/setup.ts

```diff
@@ -123,14 +124,25 @@ async function createSetupSideEffects(
   return { isIntialized: true };
 }
 
-export async function setupFleet(
-  soClient: SavedObjectsClientContract,
-  callCluster: CallESAsCurrentUser,
-  options?: { forceRecreate?: boolean }
-) {
-  // Create fleet_enroll role
-  // This should be done directly in ES at some point
-  const res = await callCluster('transport.request', {
+async function updateFleetRoleIfExists(callCluster: CallESAsCurrentUser) {
+  try {
+    await callCluster('transport.request', {
+      method: 'GET',
+      path: `/_security/role/${FLEET_ENROLL_ROLE}`,
+    });
+  } catch (e) {
+    if (e.status === 404) {
+      return;
```

Comment:
What do you think about writing this catch clause like the FTR tests? i.e. if not 404, throw e

Reply:
@jfsiii sure, would you rather I just throw the error over rejecting the promise as I do below?

Reply:
Yes, please. The `throw` becomes a `Promise.reject` in `async` functions.

So the two are effectively the same but the throw is more consistent with our existing code, like the tests in this PR.

Reply:
done